File: smart_travel_FE/lib/domain/entities/province.dart
import 'package:smart_travel/domain/entities/destinations.dart';

class ProvinceEntity {
  final int id;

  final String name;

  final String code;

  final String region;

  final String description;

  final String imageUrl;

  final bool isPopular;

  final int? destinationCount;

  final List<DestinationEntity>? destination ;

  ProvinceEntity({
    required this.id,
    required this.name,
    required this.code,
    required this.region,
    required this.description,
    required this.imageUrl,
    required this.isPopular,
    this.destination,
    this.destinationCount,
  });
}

File: smart_travel_BE/src/main/java/com/example/smart_travel_BE/entity/Province.java
package com.example.smart_travel_BE.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "provinces")
public class Province {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, unique = true, length = 100)
    private String name;

    @Column(name = "code", nullable = false, unique = true, length = 10)
    private String code;

    @Column(name = "region", length = 50)
    private String region;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "is_popular", nullable = false)
    private Boolean isPopular = false;

    @OneToMany(mappedBy = "province", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Destination> destinations = new ArrayList<>();
}